### PR TITLE
Add blacklisting wildcard support

### PR DESCRIPTION
**Pull request by DL6ER**
**By submitting this pull request, I confirm the following (please check boxes, eg [X]) _Failure to fill the template will close your PR_:**

***Please submit all pull requests against the `development` branch. Failure to do so will delay or deny your request***

- [X] I have read and understood the [contributors guide](https://github.com/pi-hole/pi-hole/blob/master/CONTRIBUTING.md).
- [X] I have checked that [another pull request](https://github.com/pi-hole/pi-hole/pulls) for this purpose does not exist.
- [X] I have considered, and confirmed that this submission will be valuable to others.
- [X] I accept that this submission may not be used, and the pull request closed at the will of the maintainer.
- [X] I give this submission freely, and claim no ownership to its content.

**How familiar are you with the codebase?:** 10

---
Add blacklisting wildcard support

```
pihole -wild google.com
pihole -wild -d google.com
```
screenshot at 2016-12-31 13-51-19

_This template was created based on the work of [`udemy-dl`](https://github.com/nishad/udemy-dl/blob/master/LICENSE)._

### CODE REVIEW

File: advanced/Scripts/list.sh [hunk 1]
```diff
@@ -47,13 +48,17 @@ helpFunc() {
 :::  -h, --help				Show this help dialog
 :::  -l, --list				Display your ${word}listed domains
 EOM
+if [[ "letter" == "b" ]]; then
```
Comment (Mcat12) [inline advanced/Scripts/list.sh]: Will this need to be `"$letter"`? I think it's currently comparing the plain string `"letter"` to `"b"`. Not a bash expert though ;)

**Comment by jpgpi250** [comment]
see my comment (question) [here](https://discourse.pi-hole.net/t/logging-for-wildcard-support/1042/8)

**Comment by DL6ER** [comment]
Processing the wildcard file for potentially blocked domains finally works!

screenshot at 2017-01-02 14-30-32

**Comment by DL6ER** [comment]
Yeah!

screenshot at 2017-01-02 14-37-38

**Comment by Mcat12** [comment]
Approved

**Comment by r0ckarong** [comment]
How would you use this from the web interface if possible? What do you need to put in the field to achieve the wildcard match?

**Comment by Mcat12** [comment]
You would put in the domain (`example.com`) and click the wildcard button. That would block `example.com` and any subdomains, such as `ads.example.com`

**Comment by guyfromhere** [comment]
Is it possible to block query-domains that contain a specific word? For example if I add `whatsapp` als wildcard-domain, everything that contain `whatsapp` is blocked?

**Comment by Mcat12** [comment]
That is not currently possible.

**Pull request merged**